README.md

```diff
@@ -29,7 +29,40 @@ It still might work on older rust versions, but this isn't ensured.
 
 The following diagram illustrates the user-facing and internal crates of the defmt framework.
 
-![defmt crates structure](assets/defmt.png)
+```mermaid
+%%{ init: { 'flowchart': { 'curve': 'step', "defaultRenderer": "elk" } } }%%
+graph TD;
+    A[[defmt]]
+    B[defmt-decoder]
+    C[defmt-itm]
+    D[defmt-macros]
+    E[defmt-parser]
+    F[defmt-print]
+    G[defmt-rtt]
+    H(defmt-semihosting)
+    I[[defmt-test]]
+    J[defmt-test-macros]
+    K[[flip-link]]
+    L[panic-probe]
+    M[[probe-run]]
+    N(qemu-run)
+
+    A --> D
+    B & D --o E
+    C & G & H & I & L --> A
+    F & M & N --o B
+    I --o J
+
+    subgraph Legend
+    O[[user facing crate]]
+    P[internal crate]
+    Q(unreleased crate)
+    O -- dependency --> P
+    Q -- pinned dependency --o P
+    end
+
+    J ~~~ Legend
```

Comment:
should this be here... what does it mean

Reply:
It is an invisible link, which is useful to change where the legend is placed. But it is actually not invisible, because the new renderer apparently does ignore quite a few things.

Reply:
https://mermaid.js.org/syntax/flowchart.html#an-invisible-link

Reply:
you may get lucky by asking at https://github.com/orgs/mermaid-js/discussions